src/nvim/version.c

```diff
@@ -1027,7 +1029,10 @@ void intro_message(int colon)
   };
 
   // blanklines = screen height - # message lines
-  blanklines = (int)Rows - (ARRAY_SIZE(lines) - 1);
+  size_t lines_size = ARRAY_SIZE(lines);
+  assert(lines_size <= LONG_MAX && Rows > 0l);
+
+  blanklines = Rows - ((int)lines_size - 1l);
```

Comment:
I think you wanted to cast to `long`, not to `int` (assertion checked long limit).